drivers/frequency/adf4030/adf4030.c

```diff
@@ -0,0 +1,1936 @@
+ * @param dev 		- The device structure.
+ * @param en	 	- The enable or disable Temperature readback feature.
+ * @return    		- 0 in case of success or negative error code.
+ */
+int adf4030_set_temperature(struct adf4030_dev *dev, bool en)
+{
+
+	int ret;
+	uint8_t val;
+	uint8_t msk;
+
+	if (!dev)
+		return -EINVAL;
+
+	if (en) {
+		ret = adf4030_spi_update_bits(dev, 0x3C, ADF4030_PD_ADC, 0x0);
+		if (ret)
+			return ret;
+
+		val = no_os_field_prep(ADF4030_ADC_CLK_SEL, false) |
+		      no_os_field_prep(ADF4030_EN_ADC_CNV, true) |
+		      no_os_field_prep(ADF4030_EN_ADC_CLK, true) |
+		      no_os_field_prep(ADF4030_EN_ADC, true);
+		msk = ADF4030_MANUAL_MODE | ADF4030_EN_ALIGN |
+		      ADF4030_EN_ADC_CLK | ADF4030_EN_ADC;
+		ret = adf4030_spi_update_bits(dev, 0x61, msk, val);
+		if (ret)
+			return ret;
+
+		return adf4030_spi_write(dev, 0x72, ADF4030_SPI_DUMMY_DATA);
+
+	} else {
+		ret = adf4030_spi_update_bits(dev, 0x61,
+					      ADF4030_EN_ADC_CNV, 0x0);
+		if (ret)
+			return ret;
+
+		return adf4030_spi_update_bits(dev, 0x3C, ADF4030_PD_ADC, 0xFF);
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Gets the value of the approximate die temperature.
+ * @param dev 		- The device structure.
+ * @param temperature		- The read value of the Temperature Readback.
+ * @return    		- 0 in case of success or negative error code.
+ */
+int adf4030_get_temperature(struct adf4030_dev *dev, int16_t *temperature)
+{
+	int ret;
+	uint8_t tmp;
+	uint8_t sign;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = adf4030_spi_read(dev, 0x93, &tmp);
+	if (ret)
+		return ret;
+
+	sign = no_os_field_get(tmp, ADF4030_TEMP_MEAS_MSB);
+
+	ret = adf4030_spi_read(dev, 0x92, &tmp);
+	if (ret)
+		return ret;
+
+	*temperature = sign ? (int16_t)tmp * -1 : tmp;
+
+	return 0;
+}
+
+/**
+ * @brief Set the chip address value.
+ * @param dev 		- The device structure.
+ * @param addr		- The desired chip address value.
+ * @return    		- 0 in case of success or negative error code.
+ */
+int adf4030_set_chip_address(struct adf4030_dev *dev, uint8_t addr)
+{
+	if (!dev)
+		return -EINVAL;
+
+	if (addr > ADF4030_CHIP_ADDRESS_MAX)
+		return -EINVAL;
+
+	dev->chip_addr = addr;
+
+	return 0;
+}
+
+/**
+ * @brief Gets the chip address value.
+ * @param dev 		- The device structure.
+ * @param addr		- The read value of the chip address.
+ * @return    		- 0 in case of success or negative error code.
+ */
+int adf4030_get_chip_address(struct adf4030_dev *dev, uint8_t *addr)
+{
+	if (!dev)
+		return -EINVAL;
+
+	*addr = dev->chip_addr;
+
+	return 0;
+}
+
+/**
+ * @brief Set the desired reference frequency and reset everything over to maximum
+ * supported value of 250MHz to the max. value and everything under the minimum
+ * supported value of 10MHz to the min.
+ * @param dev 		- The device structure.
+ * @param val		- The desired reference frequency in Hz.
+ * @return    		- 0 in case of success or negative error code.
+ */
+int adf4030_set_ref_clk(struct adf4030_dev *dev, uint32_t val)
```

Comment:
this function does not set the ref_clk just caches the value desired by the user. I'd expect here some actual operations to set the device.